replica/table.cc

```diff
@@ -405,8 +407,8 @@ table::do_add_sstable(lw_shared_ptr<sstables::sstable_set> sstables, sstables::s
     return new_sstables;
 }
 
-void table::add_sstable(sstables::shared_sstable sstable) {
-    _main_sstables = do_add_sstable(_main_sstables, std::move(sstable), enable_backlog_tracker::yes);
+void table::add_sstable(compaction_group& cg, sstables::shared_sstable sstable) {
+    cg._main_sstables = do_add_sstable(cg._main_sstables, std::move(sstable), enable_backlog_tracker::yes);
```

Comment:
This really should delegate to compaction_group so it is control of what is happening. Maybe compaction_group shouldn't let table reach into its internals (but vice versa is okay).

Reply:
Makes sense. table::add_sstable(compaction_group, ...) should become compaction_group::add_sstable(...). Same logic applying elsewhere.